Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp

```diff
@@ -497,6 +576,32 @@ void IntegrateEllipsoids::exec() {
       }
     }
   }
+
+  if (shareBackground) {
+    // loop over all bragg peaks and apply the cached background to their satellite peaks
+    for (auto it = satellitePeakMap.begin(); it != satellitePeakMap.end(); it++) {
+      for (auto satPeak = it->second.begin(); satPeak != it->second.end(); satPeak++) {
+        // subtract the cached background from the intensity
+        (*satPeak)->setIntensity((*satPeak)->getIntensity() - cachedBraggBackground[it->first].first);
+
+        // update the sigma intensity based on the new background
+        double sigInt = (*satPeak)->getSigmaIntensity();
+        (*satPeak)->setSigmaIntensity(sqrt(sigInt * sigInt + cachedBraggBackground[it->first].second));
+
+        const PeakShapeEllipsoid braggPeakShape =
+            static_cast<const PeakShapeEllipsoid &>(peaks[it->first].getPeakShape());
+        const PeakShapeEllipsoid satellitePeakShape =
+            static_cast<const PeakShapeEllipsoid &>((*satPeak)->getPeakShape());
+
+        // update the satellite's shape background radii
+        Mantid::Geometry::PeakShape_const_sptr newShape = std::make_shared<const PeakShapeEllipsoid>(
+            satellitePeakShape.directions(), satellitePeakShape.abcRadii(), braggPeakShape.abcRadiiBackgroundInner(),
+            braggPeakShape.abcRadiiBackgroundOuter(), Mantid::Kernel::QLab, "IntegrateEllipsoids");
```

Comment:
It looks like everything is being calculated correctly, but the representation in sliceviewer should be
```cpp
        Mantid::Geometry::PeakShape_const_sptr newShape = std::make_shared<const PeakShapeEllipsoid>(
            satellitePeakShape.directions(), satellitePeakShape.abcRadii(), satellitePeakShape.abcRadii(),
            satellitePeakShape.abcRadii(), Mantid::Kernel::QLab, "IntegrateEllipsoids");
```

We don't want to draw the satellite background region since it is the same as the main Bragg peak and the PeakShapeEllipsoid won't allow the background shell to have a different center than the integrated satellite ellipsoid. If we try to draw it, it will look like a different background than what was actually used for the background subtraction.

This will only be a little confusing if the user's filter out the main Bragg peaks.